cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/declarations/FunctionDeclaration.kt

```diff
@@ -111,27 +112,35 @@ open class FunctionDeclaration : ValueDeclaration(), DeclarationHolder {
             targetFunctionDeclaration.signatureTypes == signatureTypes
     }
 
-    fun hasSignature(targetSignature: List<Type>): Boolean {
-        val signature =
-            parameters
-                .stream()
-                .sorted(Comparator.comparingInt(ParameterDeclaration::argumentIndex))
-                .collect(Collectors.toList())
+    fun hasSignature(call: CallExpression): Boolean {
+        return hasSignature(call.signature, call.arguments)
+    }
+
+    // TODO: Documentation required. It's not completely clear what this method is supposed to do.
```

Comment:
In this function we have several todos and we should solve them either here if we can figure out what is relevant for documentation, if not we have to postpone it but mark it as an issue